src/seaofnodes/seaOfNodes__Translator.ml

```diff
@@ -33,50 +65,146 @@ end = struct
     let* g = Monad.State.get () in
     Monad.State.set {g with count= g.count + 1} >> Monad.State.return g.count
 
+  let get_pc () =
+    let* g = Monad.State.get () in
+    Monad.State.return @@ g.pc
+
+  let get_son () =
+    let* g = Monad.State.get () in
+    Monad.State.return @@ g.son
+
+  let set_son son = Monad.State.modify (fun g -> {g with son})
```

Comment:
We will probably only update the map, so we could provide a more precise operator here (add a node then modify the map)